docs/content/en/docs-dev/user-guide/managing-application/customizing-deployment/script-run.md

```diff
@@ -55,6 +55,51 @@ The commands run in the directory where this application configuration file exis
 
 ![](/images/script-run.png)
 
+### Execute the script file
+
+If your script is so long, you can separate the script as a file.
+You can put the file with the app.pipecd.yaml in the same dir and then you can execute the script like this.
+
+```
+apiVersion: pipecd.dev/v1beta1
+kind: KubernetesApp
+spec:
+  name: script-run
+  labels:
+    env: example
+    team: product
+  pipeline:
+    stages:
+      - name: SCRIPT_RUN
+        with:
+          run: |
+            sh script.sh
+```
+
+```
+.
+├── app.pipecd.yaml
+└── script.sh
+```
+
+## Builtin command
+
+Currently, you can use the commands which are installed in the environment for the piped.
+
+For example, If you are using the container platform and the offcial piped container image, you can use the command below.
+- git
+- ssh
+- jq
+- curl
+- and the builtin commands installed in the base image.
```

Comment:
Users can also use piped installed tools such as terraform, kubectl, helm, kustomize, etc 👀

Reply:
@khanhtc1202 
Thank you for the comment.
I think it might be unexpected usage for these tools.
WDYT?

Reply:
Sorry, I don't really get your point. Could you explain it :pray:

Reply:
📝 Discussed about it in person. I agree that users can use the commands installed for the piped.
- we can use them on the script run stage
- but this may change in the future

Reply:
fixed on [48d7819](https://github.com/pipe-cd/pipecd/pull/5127/commits/48d781964683100dcd86cc0f1dfecde6c2a572a7)

Reply:
The tool binaries in `$PIPED_TOOLS_DIR` are installed just before it's needed.
So, the binaries are sometimes absent, such as when the SCRIPT_RUN stage is the first stage of the pipeline and the binaries are not installed yet.

Reply:
I understand your both point now, and agree with @Warashi If it isn't always available, the it should not be listed. But, it could be better to mention that some tools could be provided under PIPED_TOOLS_DIR so that users can check it out by themself, wdyt? @ffjlabo @Warashi

Reply:
I agree with @khanhtc1202 .
It's okay to mention that some tools may be available under PIPED_TOOLS_DIR.
But I think we also have to mention that the tools can be absent sometimes.

Reply:
@khanhtc1202 @Warashi 
> So, the binaries are sometimes absent, such as when the SCRIPT_RUN stage is the first stage of the pipeline and the binaries are not installed yet.

I forgot it. So I think we should not explain it.

It might be confusing for users who use SCRIPT_RUN that the command does not exist until the first deployment is executed.
If the piped administrator and the app administrator are different, I don't think there is a way for the app administrator to check the dir.

WDYT?

Reply:
The app administrator can use the SCRIPT_RUN stage, such as executing `ls`, to check whether the commands exist.
But I think it's nonsense to check whether the commands exist. If the administrator has to use a command, they should install it in the SCRIPT_RUN stage or ask the piped admin to use a custom container image containing the command.

I think it's okay to mention some tools are sometimes available under PIPED_TOOLS_DIR because it's a fact.
On the other hand, users must use those tools carefully, and we must explain why users must pay attention.

Reply:
I tried adding it anyway. [bfb1765](https://github.com/pipe-cd/pipecd/pull/5127/commits/bfb176524c3e29e96c5c11ef6c870479ede2646e)

Reply:
Plz check it 🙏 @Warashi @khanhtc1202